builder/hyperv/iso/builder.go

```diff
@@ -0,0 +1,527 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+package iso
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/multistep"
+	hypervcommon "github.com/mitchellh/packer/builder/hyperv/common"
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/communicator"
+	"github.com/mitchellh/packer/helper/config"
+	"github.com/mitchellh/packer/packer"
+	powershell "github.com/mitchellh/packer/powershell"
+	"github.com/mitchellh/packer/powershell/hyperv"
+	"github.com/mitchellh/packer/template/interpolate"
+)
+
+const (
+	DefaultDiskSize = 40 * 1024        // ~40GB
+	MinDiskSize     = 256              // 256MB
+	MaxDiskSize     = 64 * 1024 * 1024 // 64TB
+
+	DefaultRamSize                 = 1 * 1024  // 1GB
+	MinRamSize                     = 32        // 32MB
+	MaxRamSize                     = 32 * 1024 // 32GB
+	MinNestedVirtualizationRamSize = 4 * 1024  // 4GB
+
+	LowRam = 256 // 256MB
+
+	DefaultUsername = "vagrant"
+	DefaultPassword = "vagrant"
```

Comment:
I dislike having defaults for these, but there are some inconsistencies concerning default values for `ssh_username` but `ssh_password` always defaults to `""`. See https://github.com/mitchellh/packer/pull/3909#discussion_r82502224

IMO default both these to `""`. Ping @mwhooker